sdk/identity/azure-identity/azure/identity/_internal/__init__.py

```diff
@@ -3,12 +3,25 @@
 # Licensed under the MIT License.
 # ------------------------------------
 import os
+from six.moves.urllib_parse import urlparse
 
 from .._constants import EnvironmentVariables, KnownAuthorities
 
 
+def normalize_authority(authority):
+    # type: (str) -> str
+    """Ensure authority has a scheme and strip trailing spaces and /"""
+
+    parsed = urlparse(authority)
+    if not parsed.scheme:
+        return "https://" + authority.rstrip(" /")
+    return authority.rstrip(" /")
```

Comment:
What if authority is an invalid url?

Reply:
I mean should we check if the url is valid, and if not, still call get_default?

Reply:
We should use a value provided by the user as best we can. Using a different one because we think we know better creates opportunities for confusing failure and undesired success.

Reply:
Agree with you. Changing host w/o user's awareness is bad. But as you see, if it is an invalid url, our code raise ServiceRequestError and user needs to check the error message to know what happened. How about raise a more meaningful type of error?

Reply:
In another word, I would prefer, if the error happens in url, we want it fail on the line of 
authority = normalize_authority(authority) if authority else get_default_authority()

Reply:
I think it's reasonable to expect users to read error messages, and focus our effort on making those useful. We can't communicate everything a user needs in the name of the exception. So, `ServiceRequestError: Invalid URL ...` seems on the mark to me. What more meaningful type do you have in mind?

Reply:
Not requests.exceptions.InvalidURL?

Reply:
No. I don't want to take a dependency on `requests` or raise a foreign exception.